src/GitHub.VisualStudio/TeamExplorer/Connect/GitHubConnectSection.cs

```diff
@@ -129,23 +182,70 @@ void UpdateConnection()
         void OnPropertyChange(object sender, System.ComponentModel.PropertyChangedEventArgs e)
         {
             if (e.PropertyName == "IsVisible" && IsVisible && View == null)
-                View = new GitHubConnectContent() { DataContext = this };
+                View = new GitHubConnectContent { DataContext = this };
+        }
+
+        void UpdateIcons(object sender, NotifyCollectionChangedEventArgs e)
+        {
+            if (e.Action == NotifyCollectionChangedAction.Add)
+            {
+                e.NewItems
+                    .Cast<ISimpleRepositoryModel>()
+                    .Select(async r =>
+                {
+                    if (String.Equals(Holder.ActiveRepo?.RepositoryPath, r.LocalPath, StringComparison.CurrentCultureIgnoreCase))
+                        SelectedRepository = r;
+                    var repo = await ApiFactory.Create(r.CloneUrl).GetRepository();
+                    r.SetIcon(repo.Private, repo.Fork);
+                }).ToList();
```

Comment:
The use of `ToList` here is confusing because we're not using the return value.

I think `ForEach` would more clearly express the intent here.

```csharp
                e.NewItems
                    .Cast<ISimpleRepositoryModel>()
                    .ForEach(async r =>
                {
                    if (String.Equals(Holder.ActiveRepo?.RepositoryPath, r.LocalPath, StringComparison.CurrentCultureIgnoreCase))
                        SelectedRepository = r;
                    var repo = await ApiFactory.Create(r.CloneUrl).GetRepository();
                    r.SetIcon(repo.Private, repo.Fork);
                });
```